### PR TITLE
cli(ticdc): support configure credentials and tls certificate via cli

### LINKED ISSUE #10636: Support client authentication by mutual tls

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #10636

### What is changed and how it works?
1. Supoort setting tls certificate via [command line, environment variables and config file](https://github.com/pingcap/tiflow/pull/10617/files#diff-7f58a572a760a24e8b6fd4b1780afeddcc1197a28778bb27fa0a0bf187440f03R268-R301).
2. Supoort configure tls certificate and client credentials via `cdc cli configure-credentials`.
3. Minor fix for [tls config of mysql connnection](https://github.com/pingcap/tiflow/pull/10617/files#diff-5f4d20fd09db3d31e541c0d2f675d46b912363864fbf78735cb488b4f5f5f16aR418-R421). we use "preferred" config here to make sure the connection is encrypted if possible, and it is consistent with the default behavior of mysql client.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
1. Set TLS Client Certificate via cli 
image
image

2. Set TiDB User Credentials via cli
image
image

### Release note 

```release-note
`Support client authentication by mutual tls`.
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: pkg/cmd/cli/cli_configure_credentials.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		}
+	default:
+		cmd.Printf("Received invalid input: %s, abort the command.\n", option)
+		return fmt.Errorf("invalid input")
+	}
+
+	return res.StoreToDefaultPath()
+}
+
+// newConfigureCredentials creates the `cli unsafe show-metadata` command.
```
Comment (flowbehappy) [inline pkg/cmd/cli/cli_configure_credentials.go:105]: Please fix the comment

**Comment by flowbehappy** [comment]
@CharlesCheung96 It is possible to add UT or  integration tests?

**Review by flowbehappy — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**